draft-kazuho-httpbis-priority.md

```diff
@@ -469,6 +474,21 @@ Element ID is interpreted as a Push ID.
 Empty:
 : A seven-bit field that has no semantic value.
 
+The HTTP/3 PRIORITY_UPDATE frame MUST NOT be sent with an invalid identifier,
+including before the request stream has been opened or before a promised
+request has been received.
+
+Because the HTTP/3 PRIORITY_UPDATE frame is sent on the control stream and
+there are no ordering guarantees between streams, a client that reprioritizes
+a request before receiving the response data might cause the server to receive
+a PRIORITY_UPDATE for an unknown request. If the request stream ID is within
+bidirectional stream limits, the PRIORITY_UPDATE frame SHOULD be buffered
+until the stream is opened and applied immediately after the request message
+has been processed. Holding PRIORITY_UPDATES consumes extra state on the peer,
+although the size of the state is bounded by bidirectional stream limits. There
+is no bound on the number of PRIORITY_UPDATES that can be sent, so an
+endpoint SHOULD store only the most recently received frame.
+
 TODO: add more description of how to handle things like receiving
```

Comment:
for server push there are different concerns, mainly because the client controls the MAX_PUSH_ID space. Imagine a client that sends MAX_PUSH_ID to MAX_INT and then issues MAX_INT copies of PRIORITY_UPDATE before any requests are sent.

I think we may want to say that the server SHOULD NOT buffer PRIORITY_UPDATES that refer to a push ID that has not been promised by the server.

Reply:
I was thinking if you receive a PRIORITY_UPDATE for a push ID that hasn't been promised, it's a fatal error?

Reply:
I added some text to clarify that you should close the connection if an un-promised Push ID is received.